dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMain.java

```diff
@@ -145,4 +77,102 @@ public static void main(String[] arguments)
       System.err.println(ex.getMessage());
     }
   }
+
+  private static ImmutableSet<LinkageProblem> runWithJarFiles(
+      LinkageCheckerArguments linkageCheckerArguments)
+      throws IOException, TransformerException, XMLStreamException {
+    ImmutableList<ClassPathEntry> inputClassPath = linkageCheckerArguments.getJarFiles();
+    ImmutableSet<ClassPathEntry> entryPoints = ImmutableSet.copyOf(inputClassPath);
+    LinkageChecker linkageChecker =
+        LinkageChecker.create(
+            inputClassPath, entryPoints, linkageCheckerArguments.getInputExclusionFile());
+    ImmutableSet<LinkageProblem> linkageProblems = linkageChecker.findLinkageProblems();
+
+    if (linkageCheckerArguments.getReportOnlyReachable()) {
+      ClassReferenceGraph graph = linkageChecker.getClassReferenceGraph();
+      linkageProblems =
+          linkageProblems.stream()
+              .filter(
+                  (LinkageProblem problem) ->
+                      graph.isReachable(problem.getSourceClass().getBinaryName()))
+              .collect(toImmutableSet());
+    }
+
+    Path writeAsExclusionFile = linkageCheckerArguments.getOutputExclusionFile();
```

Comment:
This chunk of code could be extracted to a `writeExclusionFile` helper method to avoid duplication

Reply:
Updated.